cli/src/config.rs

```diff
@@ -779,6 +779,20 @@ pub fn default_config_migrations() -> Vec<ConfigMigrationRule> {
                     .into())
             },
         ),
+        // TODO: Delete in jj 0.42.0+
+        ConfigMigrationRule::custom(
+            |layer| {
+                let Ok(Some(val)) = layer.look_up_item("git.push-new-bookmarks") else {
+                    return false;
+                };
+                val.as_bool().is_some_and(|b| b)
+            },
+            |_| {
+                Ok("`git.push-new-bookmarks` is deprecated; use \
+                    `remotes.<name>.auto-track-bookmarks` instead."
```

Comment:
`git.push-new-bookmarks` is boolean-valued but `remotes.<name>.auto-track-bookmarks` is a pattern. Can we give the user an example here? Something like this:

```rust
Ok("`git.push-new-bookmarks` is deprecated; use \
      `remotes.<name>.auto-track-bookmarks` instead.
Example: `jj config set --user remotes.origin.auto-track-bookmark 'glob:*'
".into())
```
It might be redundant with the hint for `git.auto-local-bookmark` but maybe that's okay?